test/e2e/functional/tests/e2e.go

```diff
@@ -192,4 +198,180 @@ var _ = Describe("Ingress Node Firewall", func() {
 			}, infwutils.Timeout, infwutils.Interval).Should(BeTrue(), "Failed to delete IngressNodeFirewall custom resource")
 		})
 	})
+
+	Context("Statistics", func() {
+		var config *ingressnodefwv1alpha1.IngressNodeFirewallConfig
+		var configCRExisted bool
+
+		BeforeEach(func() {
+			var err error
+			config = &ingressnodefwv1alpha1.IngressNodeFirewallConfig{}
+			err = infwutils.LoadIngressNodeFirewallConfigFromFile(config, consts.IngressNodeFirewallConfigCRFile)
+			Expect(err).ToNot(HaveOccurred())
+			config.SetNamespace(OperatorNameSpace)
+			configCRExisted = true
+			err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: config.Namespace, Name: config.Name}, config)
+			if errors.IsNotFound(err) {
+				configCRExisted = false
+				Expect(testclient.Client.Create(context.Background(), config)).Should(Succeed())
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		AfterEach(func() {
+			if !configCRExisted {
+				daemonset, err := testclient.Client.DaemonSets(config.Namespace).Get(context.Background(), consts.IngressNodeFirewallDaemonsetName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(daemonset.OwnerReferences).ToNot(BeNil())
+				Expect(daemonset.OwnerReferences[0].Kind).To(Equal("IngressNodeFirewallConfig"))
+
+				infwutils.DeleteINFConfig(config)
+			}
+		})
+
+		It("should expose at least one endpoint via a daemon metrics service", func() {
+			err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (done bool, err error) {
+				endpointSliceList, err := testclient.Client.Endpoints(OperatorNameSpace).List(context.TODO(), metav1.ListOptions{
+					LabelSelector: "app=ingress-node-firewall-daemon",
+				})
+
+				if err != nil {
+					return false, err
+				}
+
+				if len(endpointSliceList.Items) == 0 {
+					return false, nil
+				}
+
+				totalEndpoints := 0
+				for _, slice := range endpointSliceList.Items {
+					totalEndpoints += len(slice.Subsets)
+				}
+				if totalEndpoints > 0 {
+					return true, nil
+				}
+				return false, nil
```

Comment:
do we need to return err here to fail the test ?

Reply:
We want it to loop around again incase the endpoints havent updated yet.